cli/azd/pkg/apphost/generate.go

```diff
@@ -5,6 +5,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"hash/fnv"
```

Comment:
> using sha1 or sha256 and truncating the output down to 8 characters

The truncation part of it is what makes me think if using sha is worth it.  Don't we break the hash uniqueness when we reduce it to the first characters?
I was thinking in a case where 2 hashes are different from the 9th character ahead, resulting in the same kv name.

I landed it on `hash/fnv` by searching for fast and low chances of collisions.  I found this cool non crypto algorithms [analysis](https://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed) where fnv looks very good enough for what we need.

Since we are not using this for security-related stuff, I selected it over `sha+truncation`.

Reply:
> The truncation part of it is what makes me think if using sha is worth it. Don't we break the hash uniqueness when we reduce it to the first characters?
> I was thinking in a case where 2 hashes are different from the 9th character ahead, resulting in the same kv name.

My understanding is that this sort of thing is far less likely for a cryptographic hash function than a non cryptographic function - but we could compress the hash down by doing something like cutting into into blocks and xoring them together or something.

In practice, I suspect when this code moves into the CDK they will adopt something like SHA1 or SHA256.  But happy with this for now.

Reply:
> In practice, I suspect when this code moves into the CDK they will adopt something like SHA1 or SHA256. But happy with this for now.

If we go GA without moving the secure outputs strategy (kv injection) to Aspire+Cdk, they won't be able to change the naming strategy after GA, because that could break projects (or at least made them create an extra key vault and leak the previous one)

So, I created this https://github.com/Azure/azure-dev/issues/3622 issue. Ideally, we need a solid strategy for the names to GA for azd, which can be seemly transition to aspire+cdk